doc/designs/adtrust/sidconfig.md

```diff
@@ -0,0 +1,287 @@
+```
+ipa idrange-mod --rid-base INT --secondary-rid-base INT RANGENAME
+```
+
+### Mixed topology
+
+The existing server is not setup for PAC data generation. The future replica
+is installed with the latest packages, containing the updated installers.
+
+Run `ipa-replica-install`, with the ability to define:
+- NetBIOS Name: if not specified, the default value is generated from
+the IPA domain name (transform the left-most label into uppercase, keep only
+ascii characters, digits and dashes).
+- Primary RID base (default if not set: 1000)
+- Secondary RID base (default if not set: 1000000)
+- Run SIDgen task immediately: yes/no (default=no)
+
+
+## Design
+
+Installers and SID-related options:
+- the options `--add-sids`, `--netbios-name`, `--rid-base` and 
+`--secondary-rid-base` are moved from ADTrustInstallInterface to a separate
+new InstallInterface: SIDInstallInterface.
+- ADTrustInstallInterface now inherits from SIDInstallInterface.
+- `adtrustinstance.ADTRUSTInstance.__init__` now accepts an additional
+parameter: `fulltrust`. When set to True, the class ADTRUSTInstance configures
+the trust as usual. When set to False, only the SID-related part is executed.
+- `ipa-server-install` and `ipa-replica-install` now always call
+`adtrust.install_check` and `adtrust.install`, but the method is using the
+provided options (especially `options.setup_adtrust`) to know if full
+configuration is required or only the SID-related part.
+
+The `ipa-adtrust-install` code is already written in order to be
+idempotent (can be called multiple times, even with different values), and
+this property is of great value for this feature. It allows to keep
+the changes as minimal as possible to the existing installers, and
+call `ipa-adtrust-install` even if the SID-part is already setup.
+
+New command to enable SID generation after an upgrade:
+as we don't want to automatically enable SID generation during an upgrade,
+a new command is provided in order to manually enable the feature. The command
+only requires an admin ticket (no need to be root) and relies on the admin
+framework.
+
+The command uses a mechanism similar to server_conncheck:
+- the user must have Replication Administrators privilege
+- the user launches an ipa command communicating with IPA framework
+- the admin framework uses Dbus to get access a DBus interface that
+launches a command through oddjob, allowing the command to run as root.
+The oddjob command is a wrapper around `adtrustinstance.ADTRUSTInstance`.
+
+## Implementation
+
+- Dependencies: no additional dependency on ipa-server package
+- Backup and Restore: no new file to backup
+
+## Feature Management
+
+### UI
+
+No new UI for server/replica installation.
+
+`IPA server / ID ranges` tab is already available and displays Primary and 
+Secondary RID base.
+
+Open questions:
+- should we display the SID identifier / NetBIOS Name in
+`IPA Server / Configuration` tab instead of `IPA Server / Trusts / Global Trust`
```

Comment:
We can show them in both places, there is no problem with that. It would, actually, make sense to have that information available through both views.